### DISCUSSION TITLE
Why is db static?

### DISCUSSION THREAD

**Discussion by 205g0** [mikro-orm/mikro-orm#5206] [Ideas]
**Is your feature request related to a problem? Please describe.**
In Mongo and actually in general, the "database" is just a namespace and you can do a lot of sophisticated things with using different namespaces ("db") _but within  one_ project. In Mongo, there's no need to restrict the app to just one db in the init method. The concept of db is not meant to be one db per project.

Note: The underlying entities and general schema within a namespace/db are isolated as expected, so entities from namespace a cannot interact with entities from namespace b. And to simplify things and make such an architecture more meaningful the schemas of all namespaces are always equal.

**Describe the solution you'd like**
A way to set the db with each request individually. I went briefly over MikroORM (MO), so IDK how possible this is with the current architecture. 

**Describe alternatives you've considered**
Or maybe I just re-init on every request with the demanded db?  The docs do not say anything about this and this doesn't sound like an elegant solution.

Or2: I just init multiple instances of MO in an object like...
```
const dbs = {
  a: await MikroORM.init ...
  b: await MikroORM.init ...
}
```
...and access as I want, I just don't know if (haven't spent much time mit MO yet) if all the entity decorators work in a generic way with all dbs in `dbs`.

**Additional context**
Especially Mongo is great in this regard: You can just write to a non existing db and db + collection are created on the fly. This can create really powerful and DRY patterns which Mikro's static db just doesn't allow...

**Comment by B4nan** [comment]
Either use multiple ORM instances or you could use `FindOptions.schema` but that is probably not implemented in mongo. PR welcome.

**Comment by 205g0** [comment]
Ah cool, so you have this multi-tenant feature already but not for Mongo, I just found the [issue](https://github.com/mikro-orm/mikro-orm/issues/284). So, multiple instances as a work-around shouldn't hurt and I can keep the same entity declaration for all instances and do not need to duplicate them or so, right?!

Re PR, let me get a bit more familiar with MO and then why not...

**Comment by B4nan** [comment]
Multitenancy is in general not directly supported, but definitely possible. There is some work almost done in #1424, that should improve the support for SQL a lot. Supporting `FindOptions.schema` in mongo should be quite trivial I guess, but it's been some time since I used mongo myself (apart from MikroORM development).

Definitely something worth improving, it's just something I never needed myself, so don't have the experience and waiting for others to chip in.
